posthog/warehouse/models/test/test_table.py

```diff
@@ -264,3 +265,69 @@ def test_hogql_definition_nullable(self):
             table.hogql_definition().structure,
             "`id` String, `mrr` Nullable(Int64)",
         )
+
+    def test_comprehensive_table_definition(self):
+        base_columns = {
+            "int64": {"clickhouse": "Int64", "hogql": "IntegerDatabaseField"},
+            "float64": {"clickhouse": "Float64", "hogql": "FloatDatabaseField"},
+            "decimal": {"clickhouse": "Decimal(10, 2)", "hogql": "DecimalDatabaseField"},
+            "string": {"clickhouse": "String", "hogql": "StringDatabaseField"},
+            "datetime": {"clickhouse": "DateTime64(3, 'UTC')", "hogql": "DateTimeDatabaseField"},
+            "date": {"clickhouse": "Date", "hogql": "DateDatabaseField"},
+            "boolean": {"clickhouse": "Bool", "hogql": "BooleanDatabaseField"},
+            "array": {"clickhouse": "Array(String)", "hogql": "StringArrayDatabaseField"},
+            "map": {"clickhouse": "Map(String, String)", "hogql": "StringJSONDatabaseField"},
+        }
```

Comment:
Would there be a way for this list to reference the `SERIALIZED_FIELD_TO_CLICKHOUSE_MAPPING` object or schema definition so that we can ensure longer term parity. I think I'd forget to update this test upon updating the `SERIALIZED_FIELD_TO_CLICKHOUSE_MAPPING` dictionary.